Add the smothing option for 1d-marginals

Cosmicstring opened the pull request:
Following up on the discussion presented in [this](https://github.com/dfm/corner.py/issues/214) issue.

dfm commented:
It looks to me like most of the changes here aren't actually related to this new feature. Instead they revert more recent changes to the corner core. Please make sure you're working off of the current main branch, and isolate only the relevant changes. Thanks!

Cosmicstring commented:
Ah, indeed you're right. I haven't pulled for some time. 

I updated now my local repo and added the new changes, hopefully now everything is in sync with the HEAD of main.
